extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisTestUtils.scala

```diff
@@ -66,14 +66,15 @@ private[kinesis] class KinesisTestUtils extends Logging {
     new DynamoDB(dynamoDBClient)
   }
 
-  private lazy val kinesisProducer: KinesisProducer = {
-    val conf = new KinesisProducerConfiguration()
-      .setRecordMaxBufferedTime(1000)
-      .setMaxConnections(1)
-      .setRegion(regionName)
-      .setMetricsLevel("none")
+  /** Left as a protected val so that we don't need to depend on KPL outside of tests. */
+  protected val kplProducer: KinesisProducer = null
 
-    new KinesisProducer(conf)
+  private def getProducer(aggregate: Boolean): KinesisProducer = {
+    if (aggregate) {
+      kplProducer
```

Comment:
This is kind of confusing. A null-valued for `kplProducer`, which is really not used in KinesisTestUtils. A simpler approach is to use not have any `kplProducer` field. For this class, `getProducer` will either return a `KCLBasedProducer` when aggregate is false, or throw an exception saying not supported, use other class.